pkg/clusterversion/cockroach_versions.go

```diff
@@ -901,7 +901,7 @@ var rawVersionsSingleton = keyedVersions{
 // upgrade testing purposes and should never be done in real clusters;
 // 2. forced to `false` on release branches: this allows running a
 // release binary in a dev cluster.
-var developmentBranch = !envutil.EnvOrDefaultBool("COCKROACH_TESTING_FORCE_RELEASE_BRANCH", false) ||
+var developmentBranch = false ||
```

Comment:
I might be missing something -- this says "developmentBranch must true on the main development branch..." However, on `master`, we have:
```
var developmentBranch = envutil.EnvOrDefaultBool("COCKROACH_FORCE_DEV_VERSION", false)
```

Shouldn't it default to `true` on master? (looks like 5fa823b1dd05bd5480545046ebb712907086a915 is where this last changed)

I'm asking since I want to understand this better - don't think anything is necessarily wrong in this PR.